recipes/libyuv/all/conanfile.py

```diff
@@ -0,0 +1,65 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class LibyuvConan(ConanFile):
+    name = "libyuv"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://chromium.googlesource.com/libyuv/libyuv/"
+    description = "libyuv is an open source project that includes YUV scaling and conversion functionality."
+    topics = ["YUV", "libyuv", "google", "chromium"]
+    license = "BSD-3-Clause"
+    settings = "os", "compiler", "arch", "build_type"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "jpeg": [False, "libjpeg", "libjpeg-turbo"]}
+    default_options = {"shared": False,
+                       "fPIC": True,
+                       "jpeg": "libjpeg"}
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.jpeg == "libjpeg":
+            self.requires("libjpeg/9d")
+        elif self.options.jpeg == "libjpeg-turbo":
+            self.requires("libjpeg-turbo/2.0.5")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"]
+                  [self.version], destination=self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
```

Comment:
How do you let cmake know what libjpeg library you want to use?

Reply:
I'm a little confused for your question. Could you give me more info?

Reply:
You have added the option `with_jpeg` with values `False`, `libjpeg` or `libjpeg-turbo`.
But no arguments are passed to the cmake build script.
This means that the cmake script can accidentally pick up a system libjpeg implementation.

I just skimmed through its CMakeLists.txt and I can only find a single call to FindJPEG here:
https://chromium.googlesource.com/libyuv/libyuv/+/refs/heads/master/CMakeLists.txt#38

So I'm not sure whether libyuv out-of-the box supports switching libjpeg with libjpeg-turbo.

Reply:
You are right.  Could you give some suggestion to avoid pick up a system libjpeg?

Reply:
First remark is that I find it strange that they use `include(FindJPEG)` instead of `find_package(JPEG)`.
Second, using the `cmake_find_package` generator, it will include conan's generated `FindJPEG.cmake` cmake script.

I recently began using [`CMAKE_DISABLE_FIND_PACKAGE_xxxx`](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html#variable:CMAKE_DISABLE_FIND_PACKAGE_%3CPackageName%3E) to disable an optional find_package and avoid picking up a system library:
```python
if not self.options.with_libjpeg:
    self._cmake.definitions["CMAKE_DISABLE_FIND_PACKAGE_JPEG"] = True
```
(I'm not sure whether `self._cmake.definitions["CMAKE_DISABLE_FIND_PACKAGE_JPEG"] = self.options.with_libjpeg == False` works)

Reply:
Thanks for your suggestion. It helps a lot.

Reply:
Fixed